### PR TITLE
Kernel: Some more clang-tidy for FileSystem and Locking

### PR DESCRIPTION
**Pull request by Hendiadyoin1**

### CODE REVIEW

File: Kernel/Locking/Mutex.cpp [hunk 1]
```diff
@@ -323,18 +323,15 @@ void Mutex::restore_lock(Mode mode, u32 lock_count, [[maybe_unused]] LockLocatio
...
         auto previous_mode = m_mode;
-        bool need_to_block = false;
-        if (m_mode == Mode::Exclusive && m_holder != current_thread)
-            need_to_block = true;
-        else if (m_mode == Mode::Shared && (m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread)))
-            need_to_block = true;
-        if (need_to_block) {
+        if ((m_mode == Mode::Exclusive && m_holder != current_thread)
+            || (m_mode == Mode::Shared && m_shared_holders.size() != 1)
+            || !m_shared_holders.contains(current_thread)) {
```
Comment (davidot) [inline Kernel/Locking/Mutex.cpp:334]: This condition might be slightly different.
Before it is `m_mode == Mode::Shared && (m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread))`
So `m_mode == Mode::Shared` must be true and at least one of `m_shared_holders.size() != 1 || !m_shared_holders.contains(current_thread)` must be true.
But now if
`!m_shared_holders.contains(current_thread)` is true without `m_mode == Mode::Shared` being true it still takes the branch.

**Review by davidot — COMMENTED** [review]
No idea if this relevant with the CI but just spotted something which looks like a problem

File: Kernel/Locking/Mutex.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Kernel/Locking/Mutex.cpp]: Yeah that was the thing that borked CI
It made locking not working and blocked...

**Review by bgianfo — CHANGES_REQUESTED** [review]
I think these changes would be more readable if you broke out the clang tidy fixes so you had commits per individual rule, or something like that? Right now the reader is just forced to guess what you are fixing.

**Pull request merged**